ystockquote.py

```diff
@@ -13,488 +13,497 @@
+	return _request(symbol, 'v1')
 
 
 def get_eps_estimate_next_year(symbol):
-    return _request(symbol, 'e8')
+	return _request(symbol, 'e8')
 
 
 def get_holdings_value_realtime(symbol):
-    return _request(symbol, 'v7')
+	return _request(symbol, 'v7')
 
 
 def get_eps_estimate_next_quarter(symbol):
-    return _request(symbol, 'e9')
+	return _request(symbol, 'e9')
 
 
 def get_revenue(symbol):
-    return _request(symbol, 's6')
+	return _request(symbol, 's6')
 
 
 def get_book_value(symbol):
-    return _request(symbol, 'b4')
+	return _request(symbol, 'b4')
 
 
 def get_ebitda(symbol):
-    return _request(symbol, 'j4')
+	return _request(symbol, 'j4')
 
 
 def get_price_sales(symbol):
-    return _request(symbol, 'p5')
+	return _request(symbol, 'p5')
 
 
 def get_price_book(symbol):
-    return _request(symbol, 'p6')
+	return _request(symbol, 'p6')
 
 
 def get_pe(symbol):
-    return _request(symbol, 'r')
+	return _request(symbol, 'r')
 
 
 def get_pe_realtime(symbol):
-    return _request(symbol, 'r2')
+	return _request(symbol, 'r2')
 
 
 def get_peg(symbol):
-    return _request(symbol, 'r5')
+	return _request(symbol, 'r5')
 
 
 def get_price_eps_estimate_current_year(symbol):
-    return _request(symbol, 'r6')
+	return _request(symbol, 'r6')
 
 
 def get_price_eps_estimate_next_year(symbol):
-    return _request(symbol, 'r7')
+	return _request(symbol, 'r7')
 
 
 def get_short_ratio(symbol):
-    return _request(symbol, 's7')
+	return _request(symbol, 's7')
 
 
 def get_historical_prices(symbol, start_date, end_date):
-    """
-    Get historical prices for the given ticker symbol.
-    Date format is 'YYYY-MM-DD'
-
-    Returns a nested dictionary (dict of dicts).
-    outer dict keys are dates ('YYYY-MM-DD')
-    """
-    params = urlencode({
-        's': symbol,
-        'a': int(start_date[5:7]) - 1,
-        'b': int(start_date[8:10]),
-        'c': int(start_date[0:4]),
-        'd': int(end_date[5:7]) - 1,
-        'e': int(end_date[8:10]),
-        'f': int(end_date[0:4]),
-        'g': 'd',
-        'ignore': '.csv',
-    })
-    url = 'http://real-chart.finance.yahoo.com/table.csv?%s' % params
-    req = Request(url)
-    resp = urlopen(req)
-    content = str(resp.read().decode('utf-8').strip())
-    daily_data = content.splitlines()
-    hist_dict = dict()
-    keys = daily_data[0].split(',')
-    for day in daily_data[1:]:
-        day_data = day.split(',')
-        date = day_data[0]
-        hist_dict[date] = \
-            {keys[1]: day_data[1],
-             keys[2]: day_data[2],
-             keys[3]: day_data[3],
-             keys[4]: day_data[4],
-             keys[5]: day_data[5],
-             keys[6]: day_data[6]}
-    return hist_dict
+	"""
+	Get historical prices for the given ticker symbol.
+	Date format is 'YYYY-MM-DD'. Also accepts datetime
+	in addition to strings.
+
+	Returns a nested dictionary (dict of dicts).
+	outer dict keys are dates ('YYYY-MM-DD')
+	"""
+
+	if isinstance(start_date, datetime.date):
```

Comment:
im kinda not crazy about type checking here.  is there a more pythonic way to do this with duck typing and exceptions?